src/main/scala/encry/storage/levelDb/versionalLevelDB/WalletVersionalLevelDB.scala

```diff
@@ -0,0 +1,110 @@
+package encry.storage.levelDb.versionalLevelDB
+
+import cats.instances.all._
+import cats.syntax.semigroup._
+import com.google.common.primitives.Longs
+import com.typesafe.scalalogging.StrictLogging
+import encry.modifiers.state.StateModifierSerializer
+import encry.modifiers.state.box.Box.Amount
+import encry.modifiers.state.box.EncryBaseBox
+import encry.modifiers.state.box.TokenIssuingBox.TokenId
+import encry.settings.{Constants, LevelDBSettings}
+import encry.storage.levelDb.versionalLevelDB.VersionalLevelDBCompanion._
+import encry.utils.{BalanceCalculator, ByteStr}
+import encry.utils.CoreTaggedTypes.ModifierId
+import io.iohk.iodb.ByteArrayWrapper
+import org.encryfoundation.common.Algos
+import org.encryfoundation.common.utils.TaggedTypes.ADKey
+import org.iq80.leveldb.{DB, ReadOptions}
+import scorex.crypto.hash.Digest32
+
+import scala.util.Success
+
+case class WalletVersionalLevelDB(db: DB, settings: LevelDBSettings) extends StrictLogging with AutoCloseable {
+
+  import WalletVersionalLevelDBCompanion._
+
+  val levelDb: VersionalLevelDB = VersionalLevelDB(db, settings)
+
+  def getAllBoxes: Seq[EncryBaseBox] = levelDb.getAll
+      .map { case (key, bytes) => StateModifierSerializer.parseBytes(bytes, key.head) }
+      .collect {
+        case Success(box) => box
+      }
+
+  def getBoxById(id: ADKey): Option[EncryBaseBox] = {
+    levelDb.get(VersionalLevelDbKey @@ id.untag(ADKey))
+      .flatMap(wrappedBx => StateModifierSerializer.parseBytes(wrappedBx, id.head).toOption)
+  }
+
+  def getTokenBalanceById(id: TokenId): Option[Amount] = getBalances
+    .find(_._1 sameElements Algos.encode(id))
+    .map(_._2)
+
+  /**
+    * Get all elems, stored in currentVersion.
+    *
+    * @return
+    */
+  def getBoxes(qty: Int = 450): Seq[EncryBaseBox] = {
+    val readOptions = new ReadOptions()
+    readOptions.snapshot(levelDb.db.getSnapshot)
+    val result = levelDb.getCurrentElementsKeys.take(qty)
```

Comment:
I think it's better to alter `getCurrentElementsKeys` so it would take optional qty param, so we won't need to iterate over all iterator